src/main/java/com/yannbriancon/interceptor/HibernateQueryInterceptor.java

```diff
@@ -36,6 +33,7 @@
 
     private static final String HIBERNATE_PROXY_PREFIX = "org.hibernate.proxy";
     private static final String PROXY_METHOD_PREFIX = "com.sun.proxy";
+    private transient Predicate<String> filter;
```

Comment:
This should be ThreadLocal as threadQueryCount.
Otherwise, there is no way to have different filters in different threads

Reply:
True. I missed that detail